### PR TITLE
Implemented support for configuring a cluster metrics monitor to call cat/indices, and cat/shards.

### PR DESCRIPTION
**Pull request by AWSHurneyt**
### Description
Implemented support for configuring a cluster metrics monitor to call cat/indices, and cat/shards.
Documentation issue https://github.com/opensearch-project/documentation-website/issues/4437
 
### Issues Resolved
https://github.com/opensearch-project/alerting/issues/611

#### `Indices` and `Shards options appear in API selection dropdown.
Screenshot 2023-07-10 at 7 51 41 PM
#### Users can preview the response for `Indices` API. They can also, optionally, provide path parameters.
Screenshot 2023-07-10 at 7 52 06 PM
#### Users can preview the response for `Shards` API. They can also, optionally, provide path parameters.
Screenshot 2023-07-10 at 7 52 16 PM
#### An example trigger condition for the `Indices` API.
Screenshot 2023-07-10 at 7 52 33 PM
#### An example trigger condition for the `Shards` API.
Screenshot 2023-07-10 at 7 52 57 PM

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by AWSHurneyt** [comment]
The cypress test workflow is failing in this PR because it depends on the changes in PR https://github.com/opensearch-project/alerting/pull/992.

File: public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js [hunk 1]
```diff
@@ -104,6 +104,25 @@ export const API_TYPES = {
       source: 'ctx.results[0].nodes.NODE_ID.jvm.mem.heap_used_percent > 60',
     },
   },
+  CAT_INDICES: {
+    type: 'CAT_INDICES',
+    documentation: 'https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-indices/',
+    exampleText: 'index1,index2...',
+    label: 'Indices',
```
Comment (bowenlan-amzn) [inline public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js:111]: Looking at what label we are using for the cat actions
cat pending tasks: List pending tasks
cat recovery: Recovery

And now Indices, Shards, not sure if these are consistent, maybe just change all to "List ...s"

**Review by bowenlan-amzn — DISMISSED** [review]

File: public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js (hunk 1 above)
Comment (AWSHurneyt) [reply public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js:111]: Thank you! I meant to go back and change that from the place holder text. I've updated `cat/indices` and `cat/shards` to `List ...`.

**Review by lezzago — APPROVED** [review]

File: public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js [hunk 2]
```diff
@@ -104,6 +104,25 @@ export const API_TYPES = {
...
     },
   },
+  CAT_INDICES: {
+    type: 'CAT_INDICES',
+    documentation: 'https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-indices/',
+    exampleText: 'index1,index2...',
+    label: 'List indices',
+    paths: {
+      withPathParams: '_cat/indices',
+      withoutPathParams: '_cat/indices',
```
Comment (amsiglan) [inline public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js:114]: Doubt: Both should be same?

**Review by amsiglan — APPROVED** [review]

File: public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js (hunk 2 above)
Comment (AWSHurneyt) [reply public/pages/CreateMonitor/components/ClusterMetricsMonitor/utils/clusterMetricsMonitorConstants.js:114]: Yes, that's fine for now. Some of the other API use that same framework. I'll take an action item to implement regex to accomplish this; but for now, this framework supports easy extensibility.

**Pull request merged**